index.js

```diff
@@ -171,22 +160,22 @@ exports.init = function (Implementation, dependencies) {
 
           if (opts.configDir) {
             directorySmartImplementation = path.resolve('.', opts.configDir);
+
+            if (!fs.existsSync(directorySmartImplementation)) {
+              logger.error('Your Forest modelsDir option you configured does not seem to be an existing directory.');
+              return P.resolve([]);
+            }
+
+            return requireAllModels(Implementation, directorySmartImplementation);
           } else {
             directorySmartImplementation = path.resolve('.') + '/forest';
-          }
 
-          return codeSyntaxInspector
-            .extractCodeSyntaxErrorInDirectoryFile(directorySmartImplementation)
-            .then(function (hasError) {
-              if (hasError) {
-                throw new Error();
-              } else {
-                // NOTICE: Do not display an error log if the forest/ directory
-                //         does not exist.
-                return requireAllModels(Implementation,
-                  directorySmartImplementation, false);
-              }
-            });
+            if (fs.existsSync(directorySmartImplementation)) {
+              return requireAllModels(Implementation, directorySmartImplementation);
+            } else {
+              return P.resolve([]);
+            }
+          }
```

Comment:
Looks like you duplicated the logic.
the `directorySmartImplementation` value and the error log are the only things that are specific to `configDir` presence.

```javascript
var directorySmartImplementation = path.resolve('.', opts.configDir || '/forest');

if (fs.existsSync(directorySmartImplementation)) {
  return requireAllModels(Implementation, directorySmartImplementation);
} else {
  if (opts.configDir) {
    logger.error('The Forest modelsDir option you configured does not seem to be an existing directory.');
  }
  return [];
}
```
Does the same job, with less code/logic.

Reply:
`path.resolve('.') + '/forest'` is not the same as `path.resolve('.', '/forest')`

Reply:
then `path.resolve('.', 'forest')` is equivalent to `path.resolve('.') + '/forest'`